File: 0392-is-subsequence/0392-is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        idx = 0
        if(len(s) == 0):
            return True
        for i in list(t):
            if (s[idx] == i):
                idx += 1
            if idx == len(s):
                return True

        return False
        

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sIdx = 0;
        if (s.length() == 0) {
            return true;
        }
        for (int i = 0; i < t.length(); i++){
            if (s.charAt(sIdx) == t.charAt(i)){
                sIdx++;
                if (sIdx > s.length()-1){
                    return true;
                }
            }
        }
        return false;
    }
}